src/averageAges.js

```diff
@@ -55,7 +66,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people
+    .filter(mother => people.some(child => child.mother === mother.name));
+
+  const children = onlyWithSon === undefined
+    ? people.filter(child => people
+      .some(mother => mother.name === child.mother))
+
+    : people.filter(child => people.some(mother => mother.name === child.mother)
+      && child.sex === 'm');
+
+  return children
+    .map(child => child.born - mothers
+      .find(mother => mother.name === child.mother).born)
```

Comment:
Better create variable `motherBorn` for example, this way code would be easier to read.